spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -406,9 +406,11 @@ public void setPublisherReturns(boolean publisherReturns) {
 	 */
 	@Deprecated
 	public void setPublisherConfirms(boolean publisherConfirms) {
-		Assert.isTrue(!ConfirmType.SIMPLE.equals(this.confirmType),
+		Assert.isTrue(!publisherConfirms || !ConfirmType.SIMPLE.equals(this.confirmType),
 				"Cannot set both publisherConfirms and simplePublisherConfirms");
-		setPublisherConfirmType(ConfirmType.CORRELATED);
+		if (publisherConfirms) {
```

Comment:
Now let's imaging the situation that I call `setPublisherConfirms(true)` and have a `setPublisherConfirmType(ConfirmType.CORRELATED)` and then I call `setPublisherConfirms(false)`, but I don't have `setPublisherConfirmType(ConfirmType.NONE)`.

Is it possible?